### PR TITLE
feat(tests): Add test infrastructure and setup.ps1 for local setup

### PR DESCRIPTION
**Pull request by azabbasi**

### CODE REVIEW

File: sdk/iot/Azure.Iot.Hub.Service/tests/prerequisites/setup.ps1 [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+{
+    $IotHubName = $ResourceGroup
+}
+
+if (-not $AppRegistrationName)
+{
+    $AppRegistrationName = $ResourceGroup
+}
+
+$appId = az ad app list --show-mine --query "[?displayName=='$AppRegistrationName'].appId" --output tsv
```
Comment (abhipsaMisra) [inline sdk/iot/Azure.Iot.Hub.Service/tests/prerequisites/setup.ps1:65]: Question - is this for the storage service's identity based jobs API tests, or does IoT hub already have identity based authentication configured?

**Review by abhipsaMisra — APPROVED** [review]

File: sdk/iot/Azure.Iot.Hub.Service/tests/prerequisites/setup.ps1 (hunk 1 above)
Comment (azabbasi) [reply sdk/iot/Azure.Iot.Hub.Service/tests/prerequisites/setup.ps1:65]: I have not yet done any validation as to what we need the identity based auth for, this just makes sure you can setup an app registration and service principal as part of your setup or use the one you already have setup.

File: sdk/iot/test-resources.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        }
+    },
+    "variables": {
+        "iotHubResourceId": "[resourceId('Microsoft.Devices/IotHubs', parameters('baseName'))]",
+        "rbacOwnerRoleDefinitionId": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Authorization/roleDefinitions/', '8e3af657-a8ff-443c-a75c-2fe8c4bcb635')]"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Authorization/roleAssignments",
+            "apiVersion": "2018-09-01-preview",
```
Comment (prmathur-microsoft) [inline sdk/iot/test-resources.json:33]: Do we have to tie to this API version here?

**Review by prmathur-microsoft — APPROVED** [review]

File: sdk/iot/test-resources.json (hunk 2 above)
Comment (azabbasi) [reply sdk/iot/test-resources.json:33]: this is the recommended api for authorization, we can bring it down to a GA version but this is a dead simple api call and it should be safe.

**Pull request merged**